### PR TITLE
Markup `DataSourcePicker` for translations (components used in Explore only)

### LINKED ISSUE #76095: i18n: Markup `DataSourcePicker` for translations #75400

### PR DESCRIPTION
**Pull request by aditipatelpro**
**What is this feature?**

This PR adds support for i18n in one of the Explore components. This PR targets the `DataSourcePicker` component.

**Why do we need this feature?**

To add support for i18n in the `DataSourcePicker`  component.

**Who is this feature for?**

Everyone

**Which issue(s) does this PR fix?**:

Fixes #76095

**Special notes for your reviewer:**

I have added i18n support for the `DataSourcePicker` component in `ExploreToolbar` Component.

image

@ifrost  can you please provide your feedback on the PR?
 
Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by aditipatelpro** [comment]
Hi @ifrost, @JoaoSilvaGrafana, can you please review this PR?

File: public/app/features/datasources/components/picker/DataSourceModal.tsx [hunk 1]
```diff
@@ -143,7 +144,7 @@ export function DataSourceModal({
 
   return (
     <Modal
-      title="Select data source"
+      title={t('data-source-picker.modal-title', 'Select data source')}
```
Comment (JoaoSilvaGrafana) [inline public/app/features/datasources/components/picker/DataSourceModal.tsx:147]:
```suggestion
      title={t('data-source-picker.modal.title', 'Select data source')}
```

File: public/app/features/datasources/components/picker/DataSourceModal.tsx [hunk 2]
```diff
@@ -159,7 +160,7 @@ export function DataSourceModal({
           className={styles.searchInput}
           value={search}
           prefix={<Icon name="search" />}
-          placeholder="Search data source"
+          placeholder={t('data-source-picker.input-placeholder', 'Select data source')}
```
Comment (JoaoSilvaGrafana) [inline public/app/features/datasources/components/picker/DataSourceModal.tsx:163]:
```suggestion
          placeholder={t('data-source-picker.modal.input-placeholder', 'Select data source')}
```

File: public/app/features/datasources/components/picker/DataSourceModal.tsx [hunk 3]
```diff
@@ -211,7 +212,11 @@ export function DataSourceModal({
           )}
         </div>
         <div className={styles.newDSSection}>
-          <span className={styles.newDSDescription}>Open a new tab and configure a data source</span>
+          <span className={styles.newDSDescription}>
+            <Trans i18nKey="data-source-picker.configure-new-data-source">
```
Comment (JoaoSilvaGrafana) [inline public/app/features/datasources/components/picker/DataSourceModal.tsx:216]:
```suggestion
            <Trans i18nKey="data-source-picker.modal.configure-new-data-source">
```

**Review by JoaoSilvaGrafana — COMMENTED** [review]
Thanks for adding these! Just some suggestions for the keys so it is a bit more organized in the translation files 👍

File: public/app/features/datasources/components/picker/DataSourceModal.tsx (hunk 1 above)
Comment (aditipatelpro) [reply public/app/features/datasources/components/picker/DataSourceModal.tsx]: Thanks for reviewing,  I've made the changes.

File: public/app/features/datasources/components/picker/DataSourceModal.tsx (hunk 2 above)
Comment (aditipatelpro) [reply public/app/features/datasources/components/picker/DataSourceModal.tsx]: That makes sense 👍,  I've made the changes.

File: public/app/features/datasources/components/picker/DataSourceModal.tsx (hunk 3 above)
Comment (aditipatelpro) [reply public/app/features/datasources/components/picker/DataSourceModal.tsx]: That makes sense 👍,  I've made the changes.

**Review by JoaoSilvaGrafana — APPROVED** [review]
Lgtm, not sure if someone from Explore has any extra comments on this

File: public/app/features/datasources/components/picker/DataSourceList.tsx [hunk 4]
```diff
@@ -98,7 +99,9 @@ function EmptyState({ className, onClickCTA }: { className?: string; onClickCTA?
   const styles = useStyles2(getEmptyStateStyles);
   return (
     <div className={cx(className, styles.container)}>
-      <p className={styles.message}>No data sources found</p>
+      <p className={styles.message}>
+        <Trans i18nKey="data-source-picker.list.no-data-source-message"> No data sources found</Trans>
```
Comment (ifrost) [inline public/app/features/datasources/components/picker/DataSourceList.tsx:103]:
```suggestion
        <Trans i18nKey="data-source-picker.list.no-data-source-message">No data sources found</Trans>
```

extra space is probably not needed

**Review by ifrost — COMMENTED** [review]
Looking good 🙌  Could also rebase/sync with main branch? I think this might be causing the build to fail.

**Review by joshhunt — APPROVED** [review]

**Pull request merged**